### DISCUSSION TITLE
Need to clear queues(Active, Waiting, Delayed) whenever redis goes down

### DISCUSSION THREAD

**Discussion by smanchikatla152** [OptimalBits/bull#2441] [Q&A]
We are using Bull and Redis in one of our Apps and currently Bull is processing 100,000 jobs per minute.
Whenever redis goes down for a brief period, due to any issue, huge number of jobs are queued in Bull. 

Here are the approx. numbers
Active Queue: 115,000
Waiting Queue: 8,000
Delayed: 75,000

Once the Active queue has more than 100,000 jobs then processing of jobs take place very slowly. Only 500 jobs per minute are processed from then which doesn't work for us.

For Bull to process more jobs, we have to go to redis and delete all the keys that are added up and then Bull starts processing more jobs.

However, deleting the keys in redis is messing up schedules of the current/existing jobs. We have manually put all the jobs (50,000) back on schedule, which is taking huge amount of time.

Can you please suggest a solution so that Bull can process more jobs even though many jobs are queued up in Active queue due to any redis failure?

Bull version: 4.8.1

**Comment by manast** [comment]
If Redis goes down, then Bull will not be able to change the status of any job, so jobs cannot be accumulated in any state. Furthermore, the Active state is the state in which a job has been picked up by a worker and starts to be processed, you should not have more jobs in the Active state than the number of workers running times the concurrency factor you use in every worker. 
If Redis dies unexpectedly, it is possible that the workers that are currently processing jobs will fail to mark the jobs as complete, since the job lock could expire. In that case, these jobs will be moved back to wait so that another worker will process them in the future...
So all in all, I cannot provide any suggestion for the problem you are experiencing as it does not seem to be possible to happen in the scenario that you described.
